runtime/src/main/java/com/dylibso/chicory/runtime/HostGlobal.java

```diff
@@ -2,9 +2,14 @@
 
 import com.dylibso.chicory.wasm.types.MutabilityType;
 import com.dylibso.chicory.wasm.types.Value;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
 
 public class HostGlobal implements FromHost {
     private Value value;
+    private Supplier<Value> getValue;
+    private Consumer<Value> setValue;
```

Comment:
I'm not understanding why this Supplier/Consumer pair is necessary (and I find it a bit confusing) can you please explain? And if that's really necessary can we at least get rid of the fixed value field and replace it with a supplier of that value?

Reply:
It's not necessary, it's just one possible implementation to support a "shared HostGlobal".
With this design, you can easily implement a "proxy" HostGlobal that defers to some other value.
I'm expecting very little to no usage of this feature in "real world" applications TBH.

But I understand the concern, I'll try again a more "direct" approach and see how it goes 👍

Reply:
Introduced `GlobalInstance` as an indirection layer to be able to share Globals across Modules.
Let me know if this implementation looks better or feel free to propose an alternative design.